src/Stratis.Bitcoin.Features.Wallet/Helpers/SentTransactionItemModelComparer.cs

```diff
@@ -31,7 +31,10 @@ public bool Equals(TransactionItemModel x, TransactionItemModel y)
 
             bool propertiesAreEqual = (x.Id == y.Id &&
                 x.ConfirmedInBlock == y.ConfirmedInBlock &&
-                x.Type == y.Type && x.Amount == y.Amount && x.Payments.Count == y.Payments.Count);
+                x.Type == y.Type && 
+                x.Amount == y.Amount && 
+                x.Payments.Count == y.Payments.Count &&
+                x.OutputIndex == y.OutputIndex);
```

Comment:
That's not correct.
If the history items are considered different when they have different output indexes, then a coinstake transaction splitting 1 output in 2 will result in 2 history items for the split output and 1 history items for the stake.
So if you split 1000 in 500 + 500 and get 1 from staking, you'll see `receive 500`, `receive 500` and `stake +1` in the history.

What are you trying to achieve?

Reply:
@bokobza @dangershony 
This change is to fix the bug with splitting internal payments to the same address:
Maciej used a split coin to split 900 into 5 x 180. However, history was showing incorrectly:
it showed 1 x 900 as spent (correct) and 1 x 180 (incorrect as it should be 5 x 180). 

After ny discussion with @zeptin we decided to add an Output index to be able to identfy these outputs and treat them as unique. Upon inspecting the code I can see that changes in this PR will not affect history of coin split during staking, it is already catered for and Output index will not take any effect for these:

```
                        // First we look for staking transaction as they require special attention.
                        // A staking transaction spends one of our inputs into 2 outputs or more, paid to the same address.
                        if (transaction.SpendingDetails?.IsCoinStake != null && transaction.SpendingDetails.IsCoinStake.Value)
                        {
                            // We look for the output(s) related to our spending input.
                            List<FlatHistory> relatedOutputs = items.Where(h => h.Transaction.Id == transaction.SpendingDetails.TransactionId && h.Transaction.IsCoinStake != null && h.Transaction.IsCoinStake.Value).ToList();
                            if (relatedOutputs.Any())
                            {
                                // Add staking transaction details.
                                // The staked amount is calculated as the difference between the sum of the outputs and the input and should normally be equal to 1.
                                var stakingItem = new TransactionItemModel
                                {
                                    Type = TransactionItemType.Staked,
                                    ToAddress = address.Address,
                                    Amount = relatedOutputs.Sum(o => o.Transaction.Amount) - transaction.Amount,
                                    Id = transaction.SpendingDetails.TransactionId,
                                    Timestamp = transaction.SpendingDetails.CreationTime,
                                    ConfirmedInBlock = transaction.SpendingDetails.BlockHeight,
                                    BlockIndex = transaction.SpendingDetails.BlockIndex
                                };

                                transactionItems.Add(stakingItem);
                                itemsCount++;
                            }
```
Note, I removed setting of the outputindex from the above

and this one

```
                        // We don't show in history transactions that are outputs of staking transactions.
                        if (transaction.IsCoinStake != null && transaction.IsCoinStake.Value && transaction.SpendingDetails == null)
                        {
                            continue;
                        }
```

All these check happen prior to using `Distinct`